flexget/plugins/sites/rmz.py

```diff
@@ -0,0 +1,106 @@
+from flexget.utils.soup import get_soup
+from flexget.entry import Entry
+from flexget.utils.search import normalize_unicode
+
+from requests.exceptions import RequestException
+
+log = logging.getLogger('rmz')
+
+class UrlRewriteRmz(object):
+    """
+    rmz.cr (rapidmoviez.com) urlrewriter
+    Version 0.1
+    
+    Configuration
+
+    rmz:
+      filehosters_re:
+        - domain\.com
+        - domain2\.org
+
+    Only add links that match any of the regular expressions listed under url_re.
+
+    If more than one valid link is found, the url of the entry is rewritten to
+    the first link found. The complete list of valid links is placed in the
+    'urls' field of the entry.
+    
+    Therefore, it is recommended, that you configure your output to use the
+    'urls' field instead of the 'url' field.
+    
+    For example, to use jdownloader 2 as output, you would use the exec plugin:
+    exec:
+      - echo "text={{urls}}" >> "/path/to/jd2/folderwatch/{{title}}.crawljob"
+    """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'filehosters_re': {'type': 'array', 'items': {'format': 'regexp'} }
+        },
+        'additionalProperties': False
+    }
+
+    # Since the urlrewriter relies on a config, we need to create a default one
+    config = {
+        'filehosters_re': None
+    }
+
+    #grab config
+    def on_task_start(self, task, config):
+        self.config = config
+
+    # urlrewriter API
+    def url_rewritable(self, task, entry):
+        url = entry['url']
+        rewritable_regex = '^https?:\/\/(www.)?(rmz\.cr|rapidmoviez\.(com|eu))\/.*'
+        return re.match(rewritable_regex, url) is not None
+
+    @plugin.internet(log)
+   # urlrewriter API
+    def url_rewrite(self, task, entry):
+        try:
+            page = task.requests.get(entry['url'])
+        except RequestException as e:
+            raise UrlRewritingError(str(e))
+        try:
+            soup = get_soup(page.text)
+        except Exception as e:
+            raise UrlRewritingError(str(e))
+        link_elements = soup.find_all('pre', class_='links')
+        urls=[]
+        for element in link_elements:
+            urls.extend(element.text.splitlines())
+        regexps = self.config.get('filehosters_re', [])
+        if regexps:
+            log.debug('Using filehosters_re filters: %s', str(regexps))
+        else:
+            log.debug('No filehoster filters configured, using all found links.')
+        for i in reversed(range(len(urls))):
+            urls[i] = urls[i].encode('ascii', 'ignore')
+            for regexp in regexps:
+                if re.search(regexp, urls[i]):
+                    log.debug('Url: "%s" matched filehoster filter: %s', urls[i], regexp)
+                    break
+            else:
+                log.debug('Url: "%s" does not match any of the given filehoster filters: %s', urls[i], str(regexps))
+                del(urls[i])
```

Comment:
It's a bad idea to iterate over a list and delete elements from it. You should either create a copy of the list or add to another list instead of deleting.